lib/lightning_web/components/layout_components.ex

```diff
@@ -106,6 +118,16 @@ defmodule LightningWeb.LayoutComponents do
         <Icon.settings class="h-5 w-5 inline-block mr-2" />
         <span class="inline-block align-middle">Settings</span>
       </Settings.menu_item>
+      <%= for {to, icon, text, menu_item} <- @project_menu_items do %>
+        <Settings.menu_item to={to} active={@active_menu_item == menu_item}>
+          <%= Phoenix.LiveView.TagEngine.component(
+            icon,
+            [class: "h-5 w-5 inline-block mr-1"],
+            {__ENV__.module, __ENV__.function, __ENV__.file, __ENV__.line}
+          ) %>
+          <span class="inline-block align-middle"><%= text %></span>
+        </Settings.menu_item>
+      <% end %>
```

Comment:
Do we really need the icon to be dynamic like this? It feels like a slippery slope going this route.

We need to do some more abstraction here that simplifies this, @midigofrank any ideas around how we would improve the existing implementation so that we don't have to do this kind of dynamic stuff?

I'm thinking using slots would clean this up a lot, and decouple the component from using `Application.get_env` which feels like a performance liability here, ideally these extra menu items get passed in via attrs or slots...

This component was written a long long time ago, and could benefit from some `attr` specs with defaults - I see we call this whole component with `{assigns}` which is generally a no no.

Reply:
To be directly configurable it needs to be dynamic. We can try encapsulating with modules like an Extension but the code is very localized and explicit on what it does so I avoid something fancier.

Reply:
Another way when we don't want to inject dependencies would be making everything composable and `LayoutComponents.menu_items` would call set A, B or C ...